tikv/tikv
# raftstore:  support multi-files' snapshot

### What is changed and how it works?

Issue Number: close #11595

What's Changed:

- Before this PR, TiKV's only generates single SST file per cf for snapshot. This does not work well when region size is huge (such as 10GB) as large SST could lead to larger write amplification in compactions. This change would split the large snapshot into multiple files according to the sst's raw size (before compression). The size is controlled by config: server.max_snapshot_file_size. For backward compatibility consideration, the feature is only enabled in V6.1.0 or newer.  That means it will still send one snapshot file when the whole cluster version is older than v6.1.0.

- Make snapshot GC execution in the background thread


### Related changes
None

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
   Deploy older version and update one tikv with this version, the snapshot feature still works as expected in both old and new version's tikv.  And then upgrade all tikv nodes to new version and set the max snapshot file size to 50MB, the snapshot feature still works and multiple snapshot files can be observed. 

Side effects
- Breaking backward compatibility (only when incorrect config and rollback to older version, the new versions' snapshot files cannot be deleted by older version's code)

### Release note 

```release-note
TiKv supports multi-files' snapshot for large snapshot. It's only enabled in v6.1.0 or after for now. 
```
